vortex-file/src/footer/postscript.rs

```diff
@@ -39,11 +42,61 @@ impl ReadFlatBuffer for Postscript {
         fb: &<Self::Source<'buf> as Follow<'buf>>::Inner,
     ) -> Result<Self, Self::Error> {
         Ok(Self {
-            dtype: fb.dtype().map(SegmentSpec::try_from).transpose()?,
-            footer: SegmentSpec::try_from(
-                fb.footer()
-                    .ok_or_else(|| vortex_err!("Postscript missing footer segment"))?,
+            dtype: fb
+                .dtype()
+                .map(|ps| PostscriptSegment::read_flatbuffer(&ps))
+                .transpose()?,
+            statistics: fb
+                .statistics()
+                .map(|ps| PostscriptSegment::read_flatbuffer(&ps))
+                .transpose()?,
+            layout: PostscriptSegment::read_flatbuffer(
+                &fb.layout()
+                    .ok_or_else(|| vortex_err!("Postscript missing layout segment"))?,
             )?,
         })
     }
 }
+
+pub struct PostscriptSegment {
```

Comment:
How is it different than `SegmentSpec`? also should probably be `PostscriptSegmentSpec`.